api/src/main/resources/client_api.yaml

```diff
@@ -190,6 +184,36 @@ definitions:
         items:
           type: string
 
+  ResultFilters:
+    type: object
+    description: >
+      A list of filters applied to the results of a query. Only results matching the filter criteria
+      should be returned.
+    properties:
+      not:
+        description: >
+         Set to true if a result matching allOf or anyOf below should result in a result *not*
+         being returned.
+      allOf:
+        description: >
+          A list of result filters. All filters matching means a result should be returned (or not returned
+          if "not" is true.)
+        type: array
+        items:
+          $ref: "#/definitions/ResultFilters"
```

Comment:
Hopefully the various Swagger codegens can handle this properly, worth checking things look OK for Python at least.

Reply:
Will check this.

Reply:
Generated Python looks fine to me.